adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java

```diff
@@ -477,7 +478,15 @@ private AuthenticationResult acquireTokenSilentFlow(final AuthenticationRequest
             return authResult;
         }
 
-        //3. We couldn't get locally...If eligible return via broker... otherwise return local result
+        //3. try SAML Assertion
+        if (authenticationRequest.getSamlAssertion() != null) {
+            final AuthenticationResult authResultFromSaml = tryAcquireTokenSilentWithAssertion(authenticationRequest);
+            if (isAccessTokenReturned(authResultFromSaml)) {
```

Comment:
Can we add some logging here to capture both when the access token is returned and when it is not? `Logger.v` seems appropriate for the success case; `Logger.w` for the unsuccessful case

Reply:
Sure, will add it.